javascript/ql/src/semmle/javascript/frameworks/jQuery.qll

```diff
@@ -497,24 +497,11 @@ module JQuery {
       }
     }
 
-    /**
-     * Gets a node that is written to `$.fn[something]`.
-     * JQuery plugins are registered this way.
-     */
-    private DataFlow::Node getAFnWrite() {
-      exists(DataFlow::PropWrite write, DataFlow::PropRead jQueryFN |
-        write.getBase() = jQueryFN and
-        jQueryFN.getBase().getALocalSource() = JQuery::dollar() and
-        jQueryFN.getPropertyName() = "fn" and
-        result = write.getRhs()
-      )
-    }
-
     /**
      * Gets a node that is backtracked from a node written to `$.fn[something]`.
      */
     private DataFlow::SourceNode writtenToJqueryFN(DataFlow::TypeBackTracker t) {
-      t.start() and result = getAFnWrite().getALocalSource()
+      t.start() and result = any(DataFlow::Node plugin | jQueryPluginDefinition(_, plugin)).getALocalSource()
```

Comment:
Sorry. I should have pointed you here instead: https://github.com/Semmle/ql/blob/36b28386f80dc8489f6bce4ad4dad82bed2b7f83/javascript/ql/src/semmle/javascript/security/dataflow/UnsafeJQueryPluginCustomizations.qll#L88-L93